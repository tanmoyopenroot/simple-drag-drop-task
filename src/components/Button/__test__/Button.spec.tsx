import * as React from 'react';
import { mount } from 'enzyme';

import {
  Button,
  IButtonProps,
} from '../Button';

describe('Testing <Button />', () => {
  const handleOnClick = jest.fn();
  const props: IButtonProps = {
    width: '100px',
  };

  const wrapper = mount<IButtonProps, {}>(
    (
      <Button
        {...props}
        onClick={handleOnClick}
      >
        Click Me!
      </Button>
    ),
  );

  it('should be able to render with props', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper.children().length).toBe(1);
    expect(wrapper.text()).toEqual('Click Me!');
    expect(wrapper.props().width).toEqual(props.width);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be able to invoke onClick', () => {
    wrapper.simulate('click');
    expect(handleOnClick).toBeCalledTimes(1);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
