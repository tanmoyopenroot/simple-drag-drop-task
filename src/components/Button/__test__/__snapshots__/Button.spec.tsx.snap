// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Button /> should be able to invoke onClick 1`] = `"<button width=\\"100px\\" class=\\"sc-bdVaJa fstkyl\\">Click Me!</button>"`;

exports[`Testing <Button /> should be able to render with props 1`] = `"<button width=\\"100px\\" class=\\"sc-bdVaJa fstkyl\\">Click Me!</button>"`;
